### ISSUE TITLE
Hook on input tensor not called when using autograd.grad()

### ISSUE THREAD

**Issue by nanohanno** [pytorch/pytorch#35802]
## 🐛 Bug

A backward hook registered on the input Tensor of the output expression is called when gradients are calculated by running `autograd.backward(output)` but not when running `autograd.grad(output, input)`.
In both cases the gradients are calculated and are the same except that `grad()` returns a tuple. 
The documentation of `register_hook()` says 

> The hook will be called every time a gradient with respect to the Tensor is computed.

And for `autograd.grad()`

> Computes and returns the sum of gradients of outputs w.r.t. the inputs.

So is it a bug or did I misunderstand the documentation or functionalities?

## To Reproduce

```python
import torch

# backward()
def hook_backward(grad):
    print("backward hook called using backward()")
input = torch.Tensor([2])
input.requires_grad = True
input.register_hook(hook_backward)
output = 5 * input

torch.autograd.backward(output)
print(f"backward() output: {input.grad}")

# grad()
def hook_grad(grad):
    print("backward hook called using grad()")
input = torch.Tensor([2])
input.requires_grad = True
input.register_hook(hook_grad)
output = 5 * input

grad = torch.autograd.grad(output, input)
print(f"grad() output: {grad}")
```

Output:
```
backward hook called using backward()
backward() output: tensor([5.])
grad() output: (tensor([5.]),)
```

## Expected behavior

Both backward hooks are called:
```
backward hook called using backward()
backward() output: tensor([5.])
backward hook called using grad()
grad() output: (tensor([5.]),)
```

## Environment

PyTorch version: 1.4.0
Is debug build: No
CUDA used to build PyTorch: None

OS: Microsoft Windows 10 Enterprise
GCC version: Could not collect
CMake version: Could not collect

Python version: 3.7
Is CUDA available: No
CUDA runtime version: No CUDA
GPU models and configuration: No CUDA
Nvidia driver version: No CUDA
cuDNN version: No CUDA

Versions of relevant libraries:
[pip3] numpy==1.15.2
[conda] Could not collect

## Additional context

I wanted to post it in the forum but could not register or use github for login.

**Comment by sangkeun00** [comment]
This issue is still present. Can we know if anyone is working on this?

**Comment by albanD** [comment]
Hi,

No one is working on this no. But we would accept a PR fixing this!
